pkg/proto/messages/json.go

```diff
@@ -5,62 +5,83 @@
 package messages
 
 import (
-	"encoding/json"
 	"fmt"
+	"time"
+
+	"go.elastic.co/fastjson"
 )
 
-// MarshalJSON implements the JSON interface for the value type
-func (val *Value) MarshalJSON() ([]byte, error) {
+// MarshalFastJSON implements the JSON interface for the value type
+func (val *Value) MarshalFastJSON(w *fastjson.Writer) error {
```

Comment:
I can't see test coverage for any of these functions, probably because they are in a separate package from the tests in helpers.